test/correctness/simd_op_check.cpp

```diff
@@ -150,6 +150,13 @@ class SimdOpCheck : public SimdOpCheckTest {
             check("pavgb", 8 * w, u8((u16(u8_1) + u16(u8_2) + 1) >> 1));
             check("pavgw", 4 * w, u16((u32(u16_1) + u32(u16_2) + 1) / 2));
             check("pavgw", 4 * w, u16((u32(u16_1) + u32(u16_2) + 1) >> 1));
+
+            // Rounding right shifts should also use pavg
+            check("pavgb", 8 * w, u8((u16(u8_1) + 15) >> 4));
```

Comment:
I think these added checks should be in test/correctness/intrinsics.cpp instead (much lighter weight test than simd_op_check and it is target independent).

Reply:
It's a test that rounding_shift_right is both caught by pattern matching and also lowers correctly on x86, so I'll move half of it to intrinsics

Reply:
Aren't the two tests below this the same test after accounting for the pattern matching part?

Reply:
Hopefully the latest commit makes it clearer. The first two tests aren't just a rounding shift right. they strength-reduce to an average followed by a shift right. So it tests the new patterns.

The second test checks lower_rounding_shift_right does the right thing.

Reply:
Oh right, I hadn't noticed the constant isn't a rounding_shift_right. Still, I think the change you just made is good (easier tests to debug if they fail).